test/unclogged/core_test.clj

```diff
@@ -444,46 +448,64 @@
-          (is (= "only in message"
-                 (.toString ^CharArrayWriter (.getMsg syslog-message))))
-          (is (= "only in message"
-                 (.getMsgId syslog-message)))
-          (is (= "ditka"
-                 (.getAppName syslog-message)))
-          (is (= "dabears"
-                 (.getHostname syslog-message)))
-          (is (= "89"
-                 (.getProcId syslog-message)))
-          (is (= Facility/KERN
-                 (.getFacility syslog-message))) ;; instance-default
-          (is (= Severity/INFORMATIONAL
-                 (.getSeverity syslog-message))) ;; unclogged default
+          message-details {:message "only in message"
+                           :message-id "only in message"}]
+      (with-redefs [unclogged.core/make-syslog (partial fake-tcp-syslog results)]
+        (let [{:keys [syslog stream]} (c/syslog-sink conn-opts defaults)]
+          (s/put! stream message-details)
+          (let [syslog-message @(s/take! results)]
+            (is (= "only in message"
+                   (.toString ^CharArrayWriter (.getMsg syslog-message))))
+            (is (= "only in message"
+                   (.getMsgId syslog-message)))
+            (is (= "ditka"
+                   (.getAppName syslog-message)))
+            (is (= "dabears"
+                   (.getHostname syslog-message)))
+            (is (= "89"
+                   (.getProcId syslog-message)))
+            (is (= Facility/KERN
+                   (.getFacility syslog-message))) ;; instance-default
+            (is (= Severity/INFORMATIONAL
+                   (.getSeverity syslog-message))) ;; unclogged default
 
-          (is (some? syslog))
-          (is (= "localhost" (.getSyslogServerHostname syslog)))
-          (is (= 1895 (.getSyslogServerPort syslog)))
-          (is (= MessageFormat/RFC_5424 (.getMessageFormat syslog)))
-          (is (.isSsl syslog)))))))
+            (is (some? syslog))
+            (is (= "localhost" (.getSyslogServerHostname syslog)))
+            (is (= 1895 (.getSyslogServerPort syslog)))
+            (is (= MessageFormat/RFC_5424 (.getMessageFormat syslog)))
+            (is (.isSsl syslog)))))))
+  (testing "Sink setup with customizable buffer size"
+    (let [buffer-size 3
+          results (s/stream buffer-size)
+          conn-opts {:host "localhost"
+                     :port 1895
+                     :buffer-size buffer-size
+                     :transport :tls
+                     :message-format :rfc-5424}
+          defaults {:hostname "dabears"
+                    :app-name "ditka"
+                    :process-id 89
+                    :facility Facility/KERN}]
+      (with-redefs [unclogged.core/make-syslog (partial fake-tcp-syslog results)]
+        (let [{:keys [stream]} (c/syslog-sink conn-opts defaults)]
+          (is (= buffer-size (:buffer-capacity (s/description stream))))
+          (dotimes [n 2] (s/put! stream (compose-message n)))
```

Comment:
Might want to do `(dec buffer-size)` here instead of `2`.

Reply:
This is neat; didn't know about `dec`! Looks like it won't be necessary because `dotimes` executes n-1: https://clojuredocs.org/clojure.core/dotimes This wasn't failing because the test wasn't running properly due to that stream error.